zerver/views/__init__.py

```diff
@@ -409,9 +409,7 @@ def maybe_send_to_registration(request, email, full_name=''):
             # Explanation: http://stackoverflow.com/a/5605354/90777
             urllib.parse.quote_plus(full_name.encode('utf8')))))
     else:
-        return render_to_response('zerver/accounts_home.html',
-                                  {'form': form, 'current_url': request.get_full_path},
-                                  request=request)
+        return redirect('/accounts/home/?email=' + email)
```

Comment:
This code path, mainly, comes into play when the matching realm doesn't exist and the user tries to login through GitHub.